File: src/706.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class MyHashMap {
   public:
    vector<vector<pair<int, int>>> map;
    int mod;

    MyHashMap() {
        mod = 1000;
        map = vector<vector<pair<int, int>>>(mod);
    }
    void put(int key, int value) {
        int idx = key % mod;
        for (auto iterator = map[idx].begin(); iterator != map[idx].end(); ++iterator) {
            if ((*iterator).first == key) {
                (*iterator).second = value;
                return;
            }
        }
        map[idx].push_back(make_pair(key, value));
    }
    int get(int key) {
        int retVal = -1;

        int idx = key % mod;
        for (auto iterator = map[idx].begin(); iterator != map[idx].end(); ++iterator) {
            if ((*iterator).first == key) {
                retVal = (*iterator).second;
            }
        }

        return retVal;
    }
    void remove(int key) {
        int idx = key % mod;
        for (auto iterator = map[idx].begin(); iterator != map[idx].end(); ++iterator) {
            if ((*iterator).first == key) {
                map[idx].erase(iterator);
                break;
            }
        }
    }
};
/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap* obj = new MyHashMap();
 * obj->put(key,value);
 * int param_2 = obj->get(key);
 * obj->remove(key);
 */

int main(int argc, char **argv) {
    struct subject {
        vector<string> methods;
        vector<vector<int>> val;
    };
    vector<subject> testData{{{"MyHashMap", "put", "put", "get", "get", "put", "get", "remove", "get"},
                              {{}, {1, 1}, {2, 2}, {1}, {3}, {2, 1}, {2}, {2}, {2}}}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input
     *  ["MyHashMap", "put", "put", "get", "get", "put", "get", "remove", "get"]
     *  [[], [1, 1], [2, 2], [1], [3], [2, 1], [2], [2], [2]]
     *  Output
     *  [null, null, null, 1, -1, null, 1, null, -1]
     */

    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input\n";
        cout << "[";
        for (long unsigned int j = 0; j < testData[i].methods.size(); ++j) {
            cout << (j == 0 ? "" : ", ") << '"' << testData[i].methods[j] << '"';
        }
        cout << "]\n";
        cout << "[";
        for (long unsigned int j = 0; j < testData[i].val.size(); ++j) {
            cout << (j == 0 ? "" : ", ") << "[";
            for (long unsigned int k = 0; k < testData[i].val[j].size(); ++k) {
                cout << (k == 0 ? "" : ", ") << testData[i].val[j][k];
            }
            cout << "]";
        }
        cout << "]\n";

        MyHashMap cMyHashMap;
        cout << "Output\n";
        cout << "[";
        for (long unsigned int j = 0; j < testData[i].methods.size(); ++j) {
            if (testData[i].methods[j].compare("MyHashMap") == 0) {
                cout << (j == 0 ? "" : ", ") << "null";
            } else if (testData[i].methods[j].compare("put") == 0) {
                cMyHashMap.put(testData[i].val[j][0], testData[i].val[j][1]);
                cout << (j == 0 ? "" : ", ") << "null";
            } else if (testData[i].methods[j].compare("get") == 0) {
                int retVal = cMyHashMap.get(testData[i].val[j][0]);
                cout << (j == 0 ? "" : ", ") << retVal;
            } else if (testData[i].methods[j].compare("remove") == 0) {
                cMyHashMap.remove(testData[i].val[j][0]);
                cout << (j == 0 ? "" : ", ") << "null";
            }
        }
        cout << "]\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}

File: src/706.py
import sys
import logging
from typing import List, Optional
from collections import defaultdict, deque, Counter
from functools import cache, lru_cache
from bisect import bisect_left, bisect_right
from itertools import accumulate
from heapq import heapify, heappush, heappop


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class MyHashMap:
    def __init__(self):
        # 0 <= key, value <= 10^6
        self.mod = 1000
        self.map = [[-1] * self.mod for _ in range(self.mod + 1)]

    def put(self, key: int, value: int) -> None:
        row = key // self.mod
        col = key % self.mod
        self.map[row][col] = value

    def get(self, key: int) -> int:
        retVal = -1

        row = key // self.mod
        col = key % self.mod
        retVal = self.map[row][col]

        return retVal

    def remove(self, key: int) -> None:
        row = key // self.mod
        col = key % self.mod
        self.map[row][col] = -1

# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        logging.info("Input")
        for methods, val in zip([["MyHashMap", "put", "put", "get", "get", "put", "get", "remove", "get"]],
                                [[[], [1, 1], [2, 2], [1], [3], [2, 1], [2], [2], [2]]]):
            # /* Example
            #  *  Input
            #  *  ["MyHashMap", "put", "put", "get", "get", "put", "get", "remove", "get"]
            #  *  [[], [1, 1], [2, 2], [1], [3], [2, 1], [2], [2], [2]]
            #  *  Output
            #  *  [null, null, null, 1, -1, null, 1, null, -1]
            #  */
            logging.info("%s", methods)
            logging.info("%s", val)

            displayStr = []
            for key, value in zip(methods, val):
                if key == "MyHashMap":
                    obj = MyHashMap()
                    displayStr.append(None)
                elif key == "put":
                    obj.put(value[0], value[1])
                    displayStr.append(None)
                elif key == "get":
                    retVal = obj.get(value[0])
                    displayStr.append(retVal)
                elif key == "remove":
                    obj.remove(value[0])
                    displayStr.append(None)
            logging.info("Output")
            logging.info("%s", displayStr)

            print("")
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)
